File: app/src/main/java/com/imagegallery/list/ImageViewHolder.java
package com.imagegallery.list;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;

import com.imagegallery.R;
import com.squareup.picasso.Picasso;

class ImageViewHolder extends RecyclerView.ViewHolder {

    private final ImageView imageView;
    private final int imageSize;

    private final ImageAdapter.OnImageClickListener imageClickListener;

    ImageViewHolder(View itemView, ImageAdapter.OnImageClickListener imageClickListener, int imageSize) {
        super(itemView);

        this.imageSize = imageSize;
        this.imageClickListener = imageClickListener;
        imageView = (ImageView) itemView.findViewById(R.id.image);
    }

    void setImage(final ImageSearchResult image) {
        Picasso.with(imageView.getContext())
                .load(image.getUrl())
                .centerCrop()
                .resize(imageSize, imageSize)
                .into(imageView);

        imageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                imageClickListener.onImageClicked(image);
            }
        });

        imageView.setContentDescription(image.getDescription());
    }
}

File: app/src/main/java/com/imagegallery/list/ImageViewHolder.kt
package com.imagegallery.list

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView

import com.imagegallery.R
import com.imagegallery.model.PhotoSearchResultItem
import com.squareup.picasso.Picasso

internal class ImageViewHolder(
        itemView: View,
        private val imageClickListener: ImageAdapter.OnPhotoSearchResultClickListener,
        private val imageSize: Int) : RecyclerView.ViewHolder(itemView) {

    private val photoImageView: ImageView = itemView.findViewById(R.id.photoImageView) as ImageView

    fun setImage(image: PhotoSearchResultItem) {
        Picasso.with(photoImageView.context)
                .load(image.url)
                .centerCrop()
                .resize(imageSize, imageSize)
                .into(photoImageView)

        photoImageView.setOnClickListener { imageClickListener.onImageClicked(image) }
        photoImageView.contentDescription = image.title
    }
}
